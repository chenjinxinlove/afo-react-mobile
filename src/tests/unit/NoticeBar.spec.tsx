import * as Enzyme from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import NoticeBar from '../../pageages/notice-bar/';
import DemoNoticeBar from '../../pageages/notice-bar/demo';

describe('NoticeBar', () => {
  // 测试是否能正确的显示内容
  it ('should render correct contents', () => {
    const wrapper = Enzyme.shallow(<NoticeBar>test</NoticeBar>)
    expect(wrapper.find('.afo-notice-bar__content').text()).toBe('test')
  })
  // 测试不可以滚动
  it ('should not scroll', () => {
    const wrapper = Enzyme.mount(<NoticeBar scrollable={false}>test</NoticeBar>)
    expect(wrapper.find('.afo-notice-bar__content').html()).toBe('<div class="afo-notice-bar__content " style="padding-left: 0px;">test</div>')
  })
  // 测试props
  it ('should props', () => {
    const wrapper = Enzyme.mount(<NoticeBar mode="link" color="#fff" background="#ccc" fps={40} delay={10}>测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测试测</NoticeBar>)
    expect(wrapper.find('.afo-notice-bar__content').prop('style')).toEqual({"animationDelay": "10s", "animationDuration": "0s", "paddingLeft": 0})
  })
  // 测试卷快照
  it ('test snapshots', () => {
    const component = renderer.create(
      <DemoNoticeBar />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})