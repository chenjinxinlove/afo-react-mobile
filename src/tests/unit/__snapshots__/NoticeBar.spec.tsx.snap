// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoticeBar test snapshots 1`] = `
<div>
  <div
    className="afo-notice-bar"
    onClick={[Function]}
    style={
      Object {
        "background": "#fff7cc",
        "color": "#f60",
      }
    }
  >
    <div
      className="afo-notice-left-icon"
    >
      <img
        src="http://img.yzcdn.cn/public_files/2017/8/10/6af5b7168eed548100d9041f07b7c616.png"
      />
    </div>
    <div
      className="afo-notice-bar__wrap"
    >
      <div
        className="afo-notice-bar__content "
        onAnimationEnd={[Function]}
        style={
          Object {
            "animationDelay": "1s",
            "animationDuration": "0s",
            "paddingLeft": 0,
          }
        }
      >
        最近做的项目是从0开始采用Vue作为数据驱动框架。所以有很多Vue实战上的经验体会，下面就专门拿几个常用的但难度不高的VUe功能说一下
      </div>
    </div>
    
  </div>
  <div
    className="afo-notice-bar"
    onClick={[Function]}
    style={
      Object {
        "background": "#fff7cc",
        "color": "#f60",
      }
    }
  >
    
    <div
      className="afo-notice-bar__wrap"
    >
      <div
        className="afo-notice-bar__content "
        onAnimationEnd={[Function]}
        style={
          Object {
            "animationDelay": "1s",
            "animationDuration": "0s",
            "paddingLeft": 0,
          }
        }
      >
        最近做的项目是从0开始采用Vue作为数据驱动框架。所以有很多Vue实战上的经验体会，下面就专门拿几个常用的但难度不高的VUe功能说
      </div>
    </div>
    
  </div>
  <div
    className="afo-notice-bar afo-notice-bar--withicon"
    onClick={[Function]}
    style={
      Object {
        "background": "#fff7cc",
        "color": "#f60",
      }
    }
  >
    
    <div
      className="afo-notice-bar__wrap"
    >
      <div
        className="afo-notice-bar__content "
        onAnimationEnd={[Function]}
        style={
          Object {
            "animationDelay": "1s",
            "animationDuration": "0s",
            "paddingLeft": 0,
          }
        }
      >
        最近做的项目是从0开始采用Vue作为数据驱动框架。所以有很多Vue实战上的经验体会，下面就专门拿几个常用的但难度不高的VUe功能说
      </div>
    </div>
    <i
      className="afo-notice-right-icon afo-icon afo-icon-close"
      onClick={[Function]}
      style={
        Object {
          "color": undefined,
          "fontSize": "14px",
        }
      }
    />
  </div>
  <div
    className="afo-notice-bar afo-notice-bar--withicon"
    onClick={[Function]}
    style={
      Object {
        "background": "#fff7cc",
        "color": "#f60",
      }
    }
  >
    
    <div
      className="afo-notice-bar__wrap"
    >
      <div
        className="afo-notice-bar__content "
        onAnimationEnd={[Function]}
        style={
          Object {
            "animationDelay": "1s",
            "animationDuration": "0s",
            "paddingLeft": 0,
          }
        }
      >
        最近做的项目是从0开始采用Vue作为数据驱动框架。所以有很多Vue实战上的经验体会，下面就专门拿几个常用的但难度不高的VUe功能说
      </div>
    </div>
    <i
      className="afo-notice-right-icon afo-icon afo-icon-arrow"
      onClick={[Function]}
      style={
        Object {
          "color": undefined,
          "fontSize": "14px",
        }
      }
    />
  </div>
</div>
`;
